make/jdk/src/classes/build/tools/generatelsrequivmaps/EquivMapsGenerator.java

```diff
@@ -213,117 +213,133 @@ private static String[] createLangArray(int index, String[] subtags) {
+                sortedLanguageMap2.size(),
+                sortedRegionVariantMap.size());
+    }
 
-            for (String key : sortedLanguageMap1.keySet()) {
-                String value = sortedLanguageMap1.get(key);
-                writer.write("        singleEquivMap.put(\""
-                    + key + "\", \"" + value + "\");");
+    private static final String getLSRText(){
+        return  """
+                        // This is an auto-generated file and should not be manually edited.
+                        //   LSR Revision: %s
+                """.formatted(LSRrevisionDate);
+    }
+
+    private static void writeMultiEquiv(BufferedWriter writer) throws IOException {
+        for (String key : sortedLanguageMap2.keySet()) {
+            String[] values = sortedLanguageMap2.get(key);
+            if (values.length >= 2) {
+                writer.write(String.format(
+                        "        multiEquivsMap.put(\"%s\", new String[] {%s});"
+                        , key, generateValuesString(values)));
                 writer.newLine();
             }
+        }
+    }
 
+    // Use for writing an equivlancy map with one value
+    private static void writeEquiv(BufferedWriter writer, String name, Map<String, String> map) throws IOException {
+        for (String key : map.keySet()) {
+            String value = map.get(key);
+            writer.write(String.format(
+                    "        %s.put(\"%s\", \"%s\");"
+                    , name, key, value));
             writer.newLine();
-            for (String key : sortedLanguageMap2.keySet()) {
-                String[] values = sortedLanguageMap2.get(key);
-
-                if (values.length >= 2) {
-                    writer.write("        multiEquivsMap.put(\""
-                        + key + "\", new String[] {"
-                        + generateValuesString(values) + "});");
-                    writer.newLine();
-                }
-            }
+        }
+    }
 
-            writer.newLine();
-            for (String key : sortedRegionVariantMap.keySet()) {
-                String value = sortedRegionVariantMap.get(key);
-                writer.write("        regionVariantEquivMap.put(\""
-                    + key + "\", \"" + value + "\");");
-                writer.newLine();
+    private static String generateValuesString(String[] values) {
+        String outputStr = "";
+        for (int i = 0; i < values.length; i++) {
+            if (i != values.length - 1) {
+                outputStr = String.format("%s\"%s\", ", outputStr, values[i]);
+            } else {
+                outputStr = String.format("%s\"%s\"", outputStr, values[i]);
             }
 
-            writer.write(footerText);
-        } catch (IOException ex) {
-            ex.printStackTrace(System.err);
-            System.exit(1);
         }
-
+        return outputStr;
     }
 
+    private static final String footerText = "    }\n\n}";
```

Comment:
I'd prefer this one also be uppercased (and moved to a more suitable location).

Reply:
ah, missed that one, probably because of it's location too. Made uppercased and moved closer to the rest of the static text variables.